scipy/__init__.py

```diff
@@ -130,10 +130,15 @@
     from . import _distributor_init
 
     from scipy._lib import _pep440
-    if _pep440.parse(__numpy_version__) < _pep440.Version('1.16.5'):
+    # In maintenance branch, change to np_maxversion N+3 if numpy is at N
+    # See setup.py for more details
+    np_minversion = '1.16.5'
+    np_maxversion = '9.9.99'
+    if (_pep440.parse(__numpy_version__) < _pep440.Version(np_minversion) or
+            _pep440.parse(__numpy_version__) >= _pep440.Version(np_maxversion)):
         import warnings
-        warnings.warn("NumPy 1.16.5 or above is required for this version of "
-                      "SciPy (detected version %s)" % __numpy_version__,
+        warnings.warn("A NumPy version >=%s and <=%s is required for this "
+                      "version of SciPy (detected version %s)" % __numpy_version__,
```

Comment:
When I was working on backporting this, I wasn't actually sure I understood how the string substitution would get three different values for `%s` from `__numpy_version__`? Maybe I'm missing something obvious though, but you can see how I did it a bit different in #12862

Reply:
uhhh, oops - I'll forward-port your improvement